### PR TITLE
refactor(mempool): with_account_nonces to use &str,u8

### PR DESCRIPTION
**Pull request by ayeletstarkware**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1081)

### CODE REVIEW

**Review by ayeletstarkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1081)* status: 0 of 1 files reviewed, 1 unresolved discussion

___
*[`crates/mempool/src/mempool_test.rs` line 897 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1081#-O7xcCSK2i0LKCVqDDV7:-O7xcCSLDoOeq056fxl1:b-8hl5qk) ([raw file](https://github.com/starkware-libs/sequencer/blob/df789e6ae9672efd592189bb460e8ac047f56f76/crates/mempool/src/mempool_test.rs#L897)):*
> ```XML
> 
>     // Test: update through a commit block.
>     let state_changes = HashMap::from([(contract_address!(0_u8), nonce!(committed_nonce))]);
> ```

The `contract_address!` macro requires an explicit type like `_u8`. 
Another option is to unpack with `let AccountState { address, .. } = input.account_state;` and use the `address` directly instead.



_Code quote:_
```XML
u8
```

**Review by elintul — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1081)* status: all files reviewed, 1 unresolved discussion (waiting on @ayeletstarkware and @giladchase)

___
*[`crates/mempool/src/mempool_test.rs` line 897 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1081#-O7xcCSK2i0LKCVqDDV7:-O7xmVF-E2h-WdGTmeTd:bhug2gz) ([raw file](https://github.com/starkware-libs/sequencer/blob/df789e6ae9672efd592189bb460e8ac047f56f76/crates/mempool/src/mempool_test.rs#L897)):*
<details><summary><i>Previously, ayeletstarkware (Ayelet Zilber) wrote…</i></summary><blockquote>

The `contract_address!` macro requires an explicit type like `_u8`. 
Another option is to unpack with `let AccountState { address, .. } = input.account_state;` and use the `address` directly instead.
</blockquote></details>

Can you use the `commit_block` test util here (and in other account nonces <> commit block tests)?

**Review by elintul — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1081)* status: all files reviewed, 1 unresolved discussion (waiting on @ayeletstarkware and @giladchase)

___
*[`crates/mempool/src/mempool_test.rs` line 893 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/1081#-O7xzNzB7UOaewnszXFG:-O7xzNzB7UOaewnszXFH:b-4u5pe) ([raw file](https://github.com/starkware-libs/sequencer/blob/19e169d147e03c9fdc6b7bd08f1dce651d7eda92/crates/mempool/src/mempool_test.rs#L893)):*
> ```XML
> 
>     // Test: commits state change of a lower account nonce.
>     let state_changes = HashMap::from([("0x0", 0)]);
> ```

No need for hashmap, it's done inside.

**Review by ayeletstarkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1081)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @elintul and @giladchase)

___
*[`crates/mempool/src/mempool_test.rs` line 893 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/1081#-O7xzNzB7UOaewnszXFG:-O7y-TpQDWyQIVgMHFsW:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/19e169d147e03c9fdc6b7bd08f1dce651d7eda92/crates/mempool/src/mempool_test.rs#L893)):*
<details><summary><i>Previously, elintul (Elin) wrote…</i></summary><blockquote>

No need for hashmap, it's done inside.
</blockquote></details>

Done.

**Review by elintul — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1081#-:-O7y3b4pEkW6zXwk1JoT:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @giladchase)

___
*[`crates/mempool/src/mempool_test.rs` line 893 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/1081#-O7xzNzB7UOaewnszXFG:-O7y-k7s6ZvM8NBpDB0f:bo1a7bp) ([raw file](https://github.com/starkware-libs/sequencer/blob/19e169d147e03c9fdc6b7bd08f1dce651d7eda92/crates/mempool/src/mempool_test.rs#L893)):*
<details><summary><i>Previously, ayeletstarkware (Ayelet Zilber) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Summing up takeaways, so it's not repeated:
- Look for similar tests, that invoke same functionality (which you know exist), to see how the logic you're changing is invoked there.
- When using a util, check for existing usages; strive for consistency.
- Pass minimal (simplest) type possible to lessen boilerplate (conversion) code.

Please ACK this before merging.
